src/clients/java/src/main/java/com/tigerbeetle/AccountBatch.java

```diff
@@ -112,197 +118,373 @@ public byte[] getUserData() {
+     * @see <a href="https://docs.tigerbeetle.com/reference/accounts/#user_data_128">user_data_128</a>
+     */
+    public byte[] getUserData128() {
+        return getUInt128(at(Struct.UserData128));
+    }
+
+    /**
+     * @param part a {@link UInt128} enum indicating which part of the 128-bit value is to be retrieved.
+     * @return a {@code long} representing the first 8 bytes of the 128-bit value if
+     *         {@link UInt128#LeastSignificant} is informed, or the last 8 bytes if
+     *         {@link UInt128#MostSignificant}.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @see <a href="https://docs.tigerbeetle.com/reference/accounts/#user_data_128">user_data_128</a>
+     */
+    public long getUserData128(final UInt128 part) {
+        return getUInt128(at(Struct.UserData128), part);
+    }
+
+    /**
+     * @param userData128 an array of 16 bytes representing the 128-bit value.
+     * @throws IllegalArgumentException if {@code userData128} is not 16 bytes long.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     * @see <a href="https://docs.tigerbeetle.com/reference/accounts/#user_data_128">user_data_128</a>
+     */
+    public void setUserData128(final byte[] userData128) {
+        putUInt128(at(Struct.UserData128), userData128);
+    }
+
+    /**
+     * @param leastSignificant a {@code long} representing the first 8 bytes of the 128-bit value.
+     * @param mostSignificant a {@code long} representing the last 8 bytes of the 128-bit value.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     * @see <a href="https://docs.tigerbeetle.com/reference/accounts/#user_data_128">user_data_128</a>
+     */
+    public void setUserData128(final long leastSignificant, final long mostSignificant) {
+        putUInt128(at(Struct.UserData128), leastSignificant, mostSignificant);
+    }
+
+    /**
+     * @param leastSignificant a {@code long} representing the first 8 bytes of the 128-bit value.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     * @see <a href="https://docs.tigerbeetle.com/reference/accounts/#user_data_128">user_data_128</a>
+     */
+    public void setUserData128(final long leastSignificant) {
+        putUInt128(at(Struct.UserData128), leastSignificant, 0);
+    }
+
+    /**
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @see <a href="https://docs.tigerbeetle.com/reference/accounts/#user_data_64">user_data_64</a>
+     */
+    public long getUserData64() {
```

Comment:
:thinking: it feels like this API makes it too easy to ignore high order bits. Should we perhaps through an exception here if this is not the entirety of user data?